AppiumSwiftClient/common/driver.swift

```diff
@@ -9,14 +9,34 @@
 import Foundation
 import UIKit // For screenshot
 
-protocol Driver {
-    // no
+protocol DriverProtocol {
+    var currentSessionCapabilities: AppiumCapabilities { get }
+    func findElement(by locator: SearchContext, with value: String) throws -> Element
+    func findElements(by locator: SearchContext, with value: String) throws -> [Element]
+    func getCapabilities() -> [String: Any]
+    func getAvailableContexts() -> [String]
+    func getCurrentContext() -> String
+    func setContext(name: String) throws -> String
+    func getBase64Screenshot() -> String
+    func saveScreenshot(to filePath: String) -> String
+    func getPageSource() throws -> String
+    func back() throws -> String
+    func setImplicitTimeout(timeoutInMillisencods: Int) throws -> String
+    func setPageLoadTimeout(timeoutInMillisencods: Int) throws -> String
+    func setScriptTimeout(timeoutInMillisencods: Int) throws -> String
+    func setTimeout(with timeoutType: TimeoutTypesEnum, and timeoutInMilliseconds: Int) throws -> String
+    func getScreenOrientation() throws -> String
+    func rotate(to orientation: ScreenOrientationEnum) throws -> String
+    func getSettings() throws -> [String: Any]
+    func setSettings(this setting: SettingsEnum, and value: AnyValue) throws -> String
```

Comment:
Could you add a case to give a string as the setting?

I mean something like `setSettings(settings: 'settingsName', value: xxxx)`.
Because Appium Java client got issues about how to set a new setting name via the client when it had no ability to give a vanilla string as the settings name.
This may help future compatibility.

Reply:
I end up refactoring the underlying sendRequest function to accept settings as a string argument rather than overloading the method itself (would result in a lot of redundant code imo)